### PR TITLE
Adds uglify's dead_code&global_defs support for consistency with SystemJSBuilder.

### PR DESCRIPTION
**Pull request by chenzhenxi**
I can't do conditionally drop code while doing minify bundle via JSPM CLI, which SystemJsBuilder original supported.  

See  

https://github.com/systemjs/builder/blob/master/lib/output.js#L82

at line: 82  

    ast = ast.transform(uglify.Compressor({
        dead_code: true,
        global_defs: globalDefs,
        warnings: false
    }));

So I added this missing part to JSPM.

Adds uglify's dead_code and global definition support to CLI for consistency with SystemJS builder.

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -94,8 +94,8 @@ process.on('uncaughtException', function(err) {
       + '  setmode local                    Switch to locally downloaded libraries\n'
       + '  setmode remote                   Switch to CDN external package sources\n'
       + '\n'
-      + 'jspm bundle moduleA + module/b [outfile] [--minify] [--no-mangle] [--inject] [--skip-source-maps] [--source-map-contents]\n'
-      + 'jspm bundle-sfx app/main [outfile] [--format <amd|cjs|global>] \n'
+      + 'jspm bundle moduleA + module/b [outfile] [--minify] [--define <json string>] [--no-mangle] [--inject] [--skip-source-maps] [--source-map-contents]\n'
```
Comment (guybedford) [inline cli.js]: Perhaps lets call it also `global-defs`.

File: lib/build.js [hunk 2]
```diff
@@ -579,7 +579,9 @@ exports.compileDir = function(dir, options) {
 
               ast = ast.transform(uglify.Compressor({
                 warnings: false,
-                evaluate: false
+                evaluate: false,
+                dead_code: true,
+                global_defs: options.globalDefs
```
Comment (guybedford) [inline lib/build.js]: This is unrelated and can be left out.

**Comment by guybedford** [comment]
Thanks for the PR. I'm somewhat torn on this though as the build API at https://github.com/jspm/jspm-cli/blob/master/docs/api.md#class-builder is the recommended way of doing more flexible / larger builds over the CLI use case.

**Comment by chenzhenxi** [comment]
Thanks for comments.
I renamed Define to global_defines and removed changes in compile.

I thinking its convenience to have CLI to do more things.

**Comment by guybedford** [comment]
Sure thanks, I can take this but will not be able to document it as a stable API feature.
